addon/components/dialog.js

```diff
@@ -1,16 +1,24 @@
 import Component from '@glimmer/component';
 import { typeOf } from '@ember/utils';
 import { guidFor } from '@ember/object/internals';
+import { getOwnConfig } from '@embroider/macros';
 
 import { modifier } from 'ember-modifier';
 
 import { Keys } from 'ember-headlessui/utils/keyboard';
 
+function getPortalRoot() {
+  const { rootElement } = getOwnConfig();
+
+  // If we looked up a `rootElement` config at build-time, use that; otherwise use the body
+  return rootElement ? document.querySelector(rootElement) : document.body;
```

Comment:
@alexlafroscia hrm.....in our app we're seeing that this doesn't seem to be working. Just starting out trying to dive in and discover why that is, but thought I'd reach out and see if you'd noticed this not working as expected. I don't have any prior experience with the embroider macros so I'm a little out of my depth at the moment.

For clarification, we're seeing that in tests `rootElement` is undefined so the dialog is always injected in to `document.body`. Have you actually see this working in tests where it injects the dialog in to the test container instead of `document.body`?

Reply:
Yeah, this seemed to work from what I could tell... in the tests for this addon itself, this change made the dialog actually visible when testing using "dev mode". Before this PR, the dialog would end up behind the "dev mode" window and would thus be invisible...

Reply:
Yeh, I'm wondering if there might be a subtle difference between the tests in this repo vs the tests in a consuming app.

If you could verify it with your app that'd be ace. I'll keep trying to dig in from my end to try and get some more info.

Reply:
@alexlafroscia Gaaaaah!!!! You know what I think it is. I'm running the tests by going to https://localhost:4200/tests after running `ember s`. This is as opposed to running `ember t -s`.

For the most part this is ok, but there are subtle differences. And one of them is hitting me right now. The former runs with the EmberCLI `development` environment. The latter runs with the `test` environment. `rootElement` is defined only for the `test` environment. So, I imagine if I running this with `ember t -s` I'll see that it works.

Going to verify now. Apologies for a bum steer if so.

Reply:
Ah, yup, that makes sense!

I run into stuff like this with my team every so often; it's so tempting to use that `/tests` URL, but I always advise against it for reasons like this. _Usually_ the behavior in those environments is the same, but not always!